Feature/#607 typography

robbieatacato opened the pull request:
### Solving:
issue https://github.com/nl-design-system/denhaag/issues/607

- headings 1, 2, 3, 4, 5 (default, bold)
- paragraph variants: default, lead, detail

**Closing issues**

paulAcato commented on components/Typography/src/index.css:
```diff
@@ -0,0 +1,59 @@
+/* Headings */
+.denhaag-typography-h1,
```
@Robbert & @bddjong  what are your thoughts about my previous comment?

robbieatacato replied:
Good idea

bddjong replied:
@paulAcato @robbieatacato I agree. I would try to match Utrecht's bem classes however. So using `denhaag-heading-1`. See example: https://github.com/nl-design-system/utrecht/blob/main/components/heading-1/bem.scss

robbieatacato commented on components/Typography/src/index.css:
```diff
@@ -0,0 +1,59 @@
+.denhaag-typography-h4,
+.denhaag-typography-h5 {
+  font-family: var(--denhaag-typography-font-family-heading, inherit);
+  font-weight: var(--denhaag-typography-font-weight-1);
```
@paulAcato 
No, these are **_not_** referencing:  /proprietary/Proprietary/src/typography.tokens.json

But these are referencing: /proprietary/Components/src/denhaag/typography.tokens.json

Looks like it though.

paulAcato replied:
@robbieatacato , oké. But still, `font-weight-1` doesn't seem very readable and doesn't give me an impression of what it does. 

So since this is the font-weight of the heading i would suggest: `--denhaag-typography-font-weight-heading` and as fallback something like `--denhaag-typography-font-weight-base` or `--denhaag-typography-font-weight-bold`. Which are far more readable and understandable.

robbieatacato replied:
@paulAcato headings and body/paragraph text have both 2 font-weights.
So I'll name them:
 --denhaag-typography-font-weight-normal
  --denhaag-typography-font-weight-bold

bddjong commented on components/Typography/src/stories/heading1.bem.stories.mdx:
```diff
@@ -0,0 +1,41 @@
+    componentSubtitle: `${pkg.name} - ${pkg.version}`,
+    docs: { source: { type: "dynamic" } },
+    status: {
+      type: "WORK IN PROGRESS",
```
Typography is actually an `"ALPHA"` component, as you can see in the react stories. This is because Typography is already being used in an external application. This should also indicate we should be careful not to break the API unless it's necessary.

bddjong commented on proprietary/Components/src/denhaag/typography.tokens.json:
```diff
@@ -0,0 +1,36 @@
+{
+  "denhaag": {
+    "typography": {
```
Could you explain the decision to add these tokens? For the existing typography implementation, we've made a very conscious choice of using Utrecht tokens to validate their usability and make it easier to move to a NL Design System maintained typography component in the future. What options were you missing in the original implementation?

robbieatacato replied:
@bddjong 
I thought the design tokens where supposed to be added by default in: proprietary/Components/src/denhaag/.
But the values to be recycled if they already exist in denhaag.

I'll check again, to remove unnecessary tokens.

robbieatacato commented on components/Typography/src/index.css:
```diff
@@ -0,0 +1,59 @@
+  font-family: var(--denhaag-typography-font-family-heading, inherit);
+  font-weight: var(--denhaag-typography-font-weight-1);
+  line-height: var(--denhaag-typography-line-height-small);
+}
```
@paulAcato where, does '--denhaag-heading-font-size' come from?

paulAcato replied:
@robbieatacato , the tokens you define ;-) as mentioned, it was an example ;-)

bddjong reviewed (changes requested):
I feel like a lot of the changes here are duplicate or unnecessary when compared to the existing implementation of Typography in our library. Could you explain what was missing from the Typography components?

robbieatacato commented on components/Typography/src/index.css:
```diff
@@ -0,0 +1,59 @@
+  letter-spacing: var(--denhaag-typography-letter-spacing-small);
+}
+
+.denhaag-typography-p--detail-small {
```
@paulAcato the detail text is just a paragraph variant, smaller uppercase text with 2 font-size variants.

paulAcato replied:
check

robbieatacato commented on components/Typography/src/stories/heading1.bem.stories.mdx:
```diff
@@ -0,0 +1,41 @@
+<Canvas>
+  {/* eslint-disable react/no-unknown-property */}
+  <Story name="Bold">
+    <h1 class="denhaag-typography-h1 denhaag-typography--font-weight-2">
```
@paulAcato @bddjong @Robbert I did not find any use of <b> or <strong> inside the headings or paragraphs currently on the denhaag.nl site.

Although I did'nt see any class modifiers for bold text either.

bddjong replied:
Our design system doesn't support optionally bold headings. Headings are either a bold font weight, or they're not, depending on which heading level it is. I don't think we should add modifiers for the font-weight.

robbieatacato replied:
@paulAcato @bddjong @Robbert
But this is described in the design, that headings and paragraphs should have 2 font-weight variants: https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/Design-System---Gemeente-Den-Haag?node-id=3497%3A14122

This should be checked with the UX team then.

bddjong replied:
As discussed in Slack, the conclusion is that `h1`, `h4` and `h5` have to be bold, while `h2` and `h3` can support both font-weights

robbieatacato commented:
@bddjong @paulAcato I refactored the bem stories, css and removed the unnecessary design tokens: https://github.com/nl-design-system/denhaag/commit/cecbccfd15a89923f88ad4015bbfb600ef93ecc3
